lib/matplotlib/collections.py

```diff
@@ -790,29 +788,21 @@ def get_edgecolor(self):
         else:
             return self._edgecolors
 
+    def _get_default_edgecolor(self):
+        return mpl.rcParams['patch.edgecolor']
+
     def _set_edgecolor(self, c):
         set_hatch_color = True
         if c is None:
-            if (mpl.rcParams['patch.force_edgecolor'] or
-                    not self._is_filled or self._edge_default):
-                c = mpl.rcParams['patch.edgecolor']
+            if (mpl.rcParams['patch.force_edgecolor'] or self._edge_default):
+                c = self._get_default_edgecolor()
             else:
                 c = 'none'
                 set_hatch_color = False
-
-        self._is_stroked = True
-        try:
-            if c.lower() == 'none':
-                self._is_stroked = False
-        except AttributeError:
-            pass
-
-        try:
-            if c.lower() == 'face':   # Special case: lookup in "get" method.
-                self._edgecolors = 'face'
-                return
-        except AttributeError:
-            pass
+        if isinstance(c, str) and c == 'face':
```

Comment:
```suggestion
        if cbook._str_lower_equal(c, 'face'):
```

Reply:
Done, under protest; I think the explicit one-liner is not only slightly more efficient, but it is easier to read, especially for someone who does not work with the code every day, and who might therefore have to look up that function in cbook to be sure of what it actually does.